packages/salesforcedx-utils-vscode/src/cli/commandExecutor.ts

```diff
@@ -58,6 +74,64 @@ export interface CommandExecution {
   readonly stderrSubject: Observable<Buffer | string>;
 }
 
+export class CompositeCliCommandExecution implements CommandExecution {
```

Comment:
This is the wrapper execution for all the sub executions. The exit and error subjects are handled by the creator of the command and in order for the top level wrapper execution to be finished, the user must explicitly call successfulExit to emit a 0 exit code and failureExit to emit a 1 exit code, emulating what happens with a real CLI execution.